internal/cmdrpkgupdate/command.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdrpkgupdate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt/internal/errors"
+	"github.com/GoogleContainerTools/kpt/internal/util/porch"
+	porchapi "github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	command = "cmdrpkgupdate"
+)
+
+func NewCommand(ctx context.Context, rcg *genericclioptions.ConfigFlags) *cobra.Command {
+	return newRunner(ctx, rcg).Command
+}
+
+func newRunner(ctx context.Context, rcg *genericclioptions.ConfigFlags) *runner {
+	r := &runner{
+		ctx: ctx,
+		cfg: rcg,
+	}
+	r.Command = &cobra.Command{
+		Use:     "update SOURCE_PACKAGE",
+		PreRunE: r.preRunE,
+		RunE:    r.runE,
+		Hidden:  porch.HidePorchCommands,
+	}
+	r.Command.Flags().StringVar(&r.revision, "revision", "", "Revision of the upstream package to update to.")
+
+	return r
+}
+
+type runner struct {
+	ctx     context.Context
+	cfg     *genericclioptions.ConfigFlags
+	client  client.Client
+	Command *cobra.Command
+
+	revision string // Target package revision
+}
+
+func (r *runner) preRunE(cmd *cobra.Command, args []string) error {
+	const op errors.Op = command + ".preRunE"
+	client, err := porch.CreateClient(r.cfg)
+	if err != nil {
+		return errors.E(op, err)
+	}
+	r.client = client
+
+	if len(args) < 1 {
+		return errors.E(op, fmt.Errorf("SOURCE_PACKAGE is a required positional argument"))
+	}
+	if len(args) > 1 {
+		return errors.E(op, fmt.Errorf("too many arguments; SOURCE_PACKAGE is the only accepted positional arguments"))
+	}
+
+	if r.revision == "" {
+		return errors.E(op, fmt.Errorf("revision is required"))
```

Comment:
Not needed for MVP, but could we consider defaulting to latest in the future?

Reply:
Yeah, we should. I'll add a TODO on this.

Reply:
Actually, now that I have invoked this command so many times :) I would love following behavior where when we invoke `kpt alpha rpkg update <pkg>` without specifying a version, it should: list the available versions for the upstream package with last published as selected by default and if I just press enter and it runs the update with the default selected.

ofc, not scoped to this PR.